include/zephyr/debug/early_print.h

```diff
@@ -0,0 +1,26 @@
+ /*
+  * Copyright (c) 2023 Arm Limited (or its affiliates). All rights reserved.
+  *
+  * SPDX-License-Identifier: Apache-2.0
+  */
+
+#ifndef ZEPHYR_INCLUDE_DEBUG_EARLY_PRINT_H_
+#define ZEPHYR_INCLUDE_DEBUG_EARLY_PRINT_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/*
+ * Different arch has its own demand of init stages.
+ */
+#define EARLY_PRINT_INIT_0	0
+#define EARLY_PRINT_INIT_1	1
```

Comment:
Not clear to me what is the difference

Reply:
While I defined two functions something like `xxx_early_print_init` and `xxx_early_map` at the first design. But I really don't know if this is generic enough regarding all architectures, not sure if there is a platform that needs 3 or more stages to init the early print, that is why I gave an unclear stage definition here. 

I can come up with two solutions if you think the specific stage name is much better:
1) change the two definitions to `EARLY_PRINT_INIT` and `EARLY_PRINT_MAP`, and still use a single init function like what it is like now.
2) use two functions like `z_early_print_init` and `z_early_print_map`.
Any suggestions?

Reply:
Just add a comment here explaining that, that will be enough. Just to justify why we have two init levels.

Reply:
Okay, I got it, will add some explanations

Reply:
Hi Carlo, I added the comments to make it more clear. Also, change the the two definitions to EARLY_PRINT_INIT and EARLY_PRINT_MMIO_MAP, as I think it might be generic enough for all MMU system.